### ISSUE TITLE
Test phase for mapnik ebuild

### ISSUE THREAD

**Issue by xificurk** [xificurk/xificurk-gentoo-overlay#1]
Figure out how to incorporate test phase for mapnik ebuild.

**Comment by jlpoolen** [comment]
I updated the following wiki with steps I took for testing using the tests provided in the build: http://wiki.openstreetmap.org/wiki/Deploying_your_own_Slippy_Map_Gentoo#Mapnik

When I tested, the test passed the first stage "visual test", but then failed elsewhere.  I know from doing a hand built installation that when the testing completes, an HTML page is generated that displays the expected and generated PNGs side by side for visual confirmation -- this is really helpful.

![2013_10_09_12_14_49_root_themis_usr_local_src](https://f.cloud.github.com/assets/227313/1300769/66f5e36a-3117-11e3-8acb-83d91560019a.png)

**Comment by jlpoolen** [comment]
The above compilation of mapnik was using Python 2.7:

```
 FEATURES="keepwork" emerge sci-geosciences/mapnik
```

As an experiment, I changed the system environment's Python to 3.2 and successfully compiled mapnik again.  (Note: when using the "keepwork" feature, you have to manually erase the prior build, otherwise make will use what already was compiled.).  I then went to test within the 3.2 built tree and the test errored out, more so than in the 2.7 build. 

```
    themis mapnik-v2.2.0 # make test
    *** Running visual tests...
      File "tests/visual_tests/test.py", line 5
        mapnik.logger.set_severity(mapnik.severity_type.None)
                                                ^
    SyntaxError: invalid syntax
    *** Running C++ tests...
    ..clipping_test-bin: /usr/include/boost/detail/lightweight_test.hpp:49: boost::detail::report_errors_reminder::~report_errors_reminder(): Assertion `called_report_errors_function' failed.
    ./run_tests: line 10: 23235 Aborted                 ${FILE} -q -d .
    ..........
    *** Running python tests...
      File "tests/run_tests.py", line 7
        except ImportError, e:
                      ^
    SyntaxError: invalid syntax
    make: *** [test] Error 136
    themis mapnik-v2.2.0 # 

themis mapnik-v2.2.0 # eselect python list
Available Python interpreters:
  [1]   python2.7
  [2]   python3.2 *
themis mapnik-v2.2.0 # eselect python set 1
themis mapnik-v2.2.0 # make test
```

The above error looked like a Python incompatability issue (2.7 code running in a 3.2 environment), so I change the environment back to 2.7 and proceeded to "make test" within the tree built under Python 3.2.  I got as far as I did when I performed a "make test" within a tree built on Python 2.7 (see screen shot above in prior comment).

Therefore, I'm concluding that mapnik 2.2 will build in Gentoo under either 2.7 or 3.2, but to run the built-in tests, you have to be running Python 2.7, and even then, the tests may not complete.  The next step will be to study the testing scripts and look for the bugs cause the test to eventually fail.  At least the image generation portion of the test appears to be working.

**Comment by xificurk** [comment]
Yeah, the incompatibilities between Python 2.7 and 3.2 are not that surprising. Python 3 support is still pretty new feature (in mapnik context), AFAIK the code (which means tests as well) is still in 2.x style and the support for Python 3 is added by running 2to3 script on the relevant parts.

**Comment by jlpoolen** [comment]
There are three tests that "make test" invokes.  The first and third may be called individually with Python, the second test "C++" is shell script driven.

I mentioned earlier an HTML report that show side by side the PNG and the generated PNG.  https://github.com/mapnik/mapnik/blob/master/tests/visual_tests/test.py @285 shows us that the HTML is only generated if there are errors.  In my case with mapnik built through this project's most excellent ebuild, all the PNG tests were successful, so no HTML report was generated.  Previously, when I built mapnik by hand, I had a nasty soft link (an errant hack from another project) which mixed the 32 and 64 bit branches so my PNG tests were failing, then I saw mention of the error report.  The error facility really should be documented rather than leaving it for someone like me to dig around and basically document portions of it.

I still do not know with "NO-OP ..." stands for, but it doesn't seem to be an error condition, perhaps it is related to something unique to a particular operating system, e.g. in Linux, the test is skipped?

The third and last test is called with the "-q " flag [quiet]; if you invoke the test without the -q, then you get reasonable status messages for each test.  (Using the -v [verbose] flag really outputs a tremendous amount of detail.).  So, above, I failed the 2nd and 3rd tests.

Test 2 is invoked with a shell script loop... probably best to leave it as it is, though the heart of the test is:

```
 ${FILE} -q -d .;
```

I'm not familiar with C++ to know what the "-q" and "-d" flags are for.

Test 3 "Python" may be invoked without the -q flag:

```
 themis mapnik-v2.2.0 # python tests/run_tests.py
```

In my test, the last several lines were:
   python_tests.geojson_plugin_test.test_geojson_properties ... ok
   python_tests.geojson_plugin_test.test_that_nonexistant_query_field_throws ... ok
   python_tests.geometry_io_test.test_wkb_parsing ... Killed

So it looks like the test_wkb (parsing shapes and json) is the one failing.
